### PR TITLE
[WIP] travis: update matrix to test current versions of Elastica

### PR DESCRIPTION
**Pull request by TomasVotruba**
Ref #224, I deleted old PR during cleanup, so this is a replacement PR only.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -1,21 +1,43 @@
...
-  - ES_VERSION=0.90.2 ES_TEST_HOST=http://localhost:9200
-  - ES_VERSION=0.90.3 ES_TEST_HOST=http://localhost:9200
-  - ES_VERSION=0.90.4 ES_TEST_HOST=http://localhost:9200
-  - ES_VERSION=0.90.5 ES_TEST_HOST=http://localhost:9200
+  global:
+    - ES_VERSION="1.7.1"
+    - ES_TEST_HOST=http://localhost:9200
+
+before_install:
+  - ./travis/download_run_es.sh
```
Comment (ruflin) [inline .travis.yml]: @TomasVotruba The file seems to be called ./travis/download_and_run_es.sh (the and is missing). That is why all travis builds already fail at the moment.

**Comment by ruflin** [comment]
@TomasVotruba Let me know if I can help somehow to get this to "green". I also refactored the way Elastica does the build on travis. Perhaps some stuff can be copied from there: https://github.com/ruflin/Elastica

**Comment by polyfractal** [comment]
@TomasVotruba  Ok, just merged this after fixing up a few conflicts.  Will keep working on it until we get Travis up to speed, thanks for PR and all the effort :)

In the future I want to explore the docker layout that @ruflin is using, but for now, let's just get basic Travis running :)

**Comment by polyfractal** [comment]
Heh.  This is why I procrastinated working on Travis for so long...it's never easy to get Travis to play nicely with our integration matrix:

image

But now it works!

screen shot 2015-11-06 at 5 52 54 pm

HHVM is still broken for now, some syntax in the yaml test runner itself is making HHVM unhappy.  I'm having difficulties installing HHVM on my laptop, but once I sort that out, tests should go green shortly.  I don't believe there is anything in the client itself that isn't compatible with HHVM.

So, yay!  I'll get badges up soon, and merge this stuff over to 2.0.  Thanks for the prodding to get this in motion :)

**Comment by TomasVotruba** [comment]
Wow, that's great job. The matrix looks very well. HHVM is failing for me on many levels as well It has it's edge cases. So this is really great success for me.

Thanks for patience and finalizing this up!

**Comment by ruflin** [comment]
:+1:

**Comment by TomasVotruba** [comment]
@ruflin Good idea. #325

**Pull request merged**